openmc/deplete/microxs.py

```diff
@@ -199,17 +206,37 @@ def from_csv(cls, csv_file, **kwargs):
         if 'float_precision' not in kwargs:
             kwargs['float_precision'] = 'round_trip'
 
-        micro_xs = cls(read_csv(csv_file, index_col=0, **kwargs))
+        df = pd.read_csv(csv_file, **kwargs)
+        df.set_index(['nuclides', 'reactions', 'groups'], inplace=True)
+        nuclides = list(df.index.unique(level='nuclides'))
+        reactions = list(df.index.unique(level='reactions'))
+        groups = list(df.index.unique(level='groups'))
+        shape = (len(nuclides), len(reactions), len(groups))
+        data = df.values.reshape(shape)
+        return cls(data, nuclides, reactions)
 
-        cls._validate_micro_xs_inputs(list(micro_xs.index),
-                                      list(micro_xs.columns),
-                                      micro_xs.to_numpy())
-        return micro_xs
+    def __getitem__(self, index):
+        nuc, rx = index
+        i_nuc = self.nuclides.index(nuc)
+        i_rx = self.reactions.index(rx)
+        return self.data[i_nuc, i_rx]
```

Comment:
The default in numpy is that if you don't specify the trailing dimensions, they are assumed to be full, so this would really be equivalent to `self.data[i_nuc, i_rx, :]`

Reply:
Gotcha